### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by tribaeka**

### CODE REVIEW

File: app/core/service/course/course.service.js [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
+        function addCourse(data, array) {
+            array.unshift(data);
+        }
+        function editCourse(data, array) {
+            var course = {
+                title: data.title,
+                description: data.description,
+                uploadDate: data.uploadDate
+            };
+            this.addDisplayDateAndTimeAfterUpdating(course);
```
Comment (kirillparamonov) [inline app/core/service/course/course.service.js]: It looks strange for me. Are you sure that it works in the same way as it's expected?

File: app/core/service/course/course.service.js [hunk 2]
```diff
@@ -1,40 +1,51 @@
...
-        .factory('courseService', ['$http', function($http) {
-            function loadCourses() {
-                return $http.get('data/courses.json');
-            }
+angular
+    .module('course')
+    .factory('courseService', [ '$http', function($http) {
+        function loadCourses() {
+            return $http.get('data/courses.json');
+        }
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/service/course/course.service.js]: done

File: app/core/service/course/course.service.js (hunk 1 above)
Comment (tribaeka) [reply app/core/service/course/course.service.js]: removed

File: app/core/component/course-list/course-list.template.html [hunk 3]
```diff
@@ -1,25 +1,29 @@
...
                 </button>
-                <button type="button" class="btn btn-delete" ng-click="$ctrl.deleteCourse(course)">
+                <button type="button" class="btn btn-delete action-btn" ng-click="$ctrl.deleteCourse(course)">
                     <span>Delete</span>
                 </button>
             </div>
         </li>
     </ul>
     <button type="button" class="btn btn-secondary "
             ng-click="$ctrl.increase()">LOAD MORE</button>
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/component/course-list/course-list.template.html]: renamed

File: app/core/component/app-header/app-header.controller.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+            console.log('on init app-header');
+        };
+        $ctrl.$onDestroy = function() {
+            console.log('on destroy app-header');
+        };
+        $ctrl.$onChanges = function(changes) {
+            console.log('on changes');
+        };
+        userService.loadUser().then(function(response) {
+            $ctrl.user = response.data;
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/component/app-header/app-header.controller.js]: renamed

File: app/core/service/course/course.service.js [hunk 5]
```diff
@@ -1,40 +1,51 @@
...
-        .module('course')
-        .factory('courseService', ['$http', function($http) {
-            function loadCourses() {
-                return $http.get('data/courses.json');
-            }
+angular
+    .module('course')
+    .factory('courseService', [ '$http', function($http) {
+        function loadCourses() {
+            return $http.get('data/courses.json');
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/service/course/course.service.js]: done

File: app/core/component/forms/forms.controller.js [hunk 6]
```diff
@@ -1,48 +1,49 @@
...
-                    uploadDate: new Date().toISOString()
-                };
+angular
+    .module('forms')
+    .controller('formsController', function($rootScope) {
+        var $ctrl = this;
+        var toggleFromForAddEvent = 'toggleFromForAdd';
+        var addCourseEvent = 'addCourse';
+        var editCourseEvent = 'editCourse';
+        var pushCourseToEditFormEvent = 'pushCourseToEditForm';
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/component/forms/forms.controller.js]: 2/4 renamed

**Pull request merged**